### PR TITLE
Add observability to TransactionKeyValueServiceManagerFactory.

### PR DESCRIPTION
**Pull request by jkozlowski**
## General
**Before this PR**:
Cannot share dialogue threads or have observability.

**After this PR**:

==COMMIT_MSG==
Add observability to TransactionKeyValueServiceManagerFactory.
==COMMIT_MSG==

**Priority**:

**Concerns / possible downsides (what feedback would you like?)**:

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:
Breaks API we are iterating on.

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:
No

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:
Yes.

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:
No

**Does this PR need a schema migration?**
No.

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java [hunk 1]
```diff
@@ -412,7 +413,7 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
 
         Refreshable<SweepConfig> sweepConfig = runtime.map(AtlasDbRuntimeConfig::sweep);
 
-        KeyValueService keyValueService = initializeCloseable(
+        KeyValueService internalKeyValueService = initializeCloseable(
```
Comment (jkozlowski) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:416]: Driveby: I think this is the line we should hold, and eventually this will be the usage, only for internal tables.

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java [hunk 2]
```diff
@@ -621,7 +622,17 @@ private <T> TransactionKeyValueServiceManager createTransactionKeyValueServiceMa
                         metricsManager,
                         config().initializeAsync());
         return factory.create(
-                coordinationService, keyValueServiceManager, config, runtimeConfigRefreshable, initializeAsync);
+                config().namespace()
+                        .orElseThrow(() -> new SafeIllegalArgumentException(
+                                "Namespace must be set when using transactionKeyValueService")),
+                reloadingFactory()
```
Comment (jkozlowski) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:628]: @jeremyk-91 we should clean this up, there's quite a few places where UserAgent is passed explicitly, where you could really just hide it inside the ReloadingFactory.

Comment (jeremyk-91) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:628]: yeah this is probably from the pre-dialogue move

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java (hunk 1 above)
Comment (jeremyk-91) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:416]: This is correct here, but for the future: please be careful with this kind of rename as a drive-by, reviewing this took more time than the "main" part of this PR!

I think you are doing this, but because some of the classes are used by consumers who are still directly creating their stuff differently we need to be careful about sending the wrong message. This path isn't used by the main alternative consumer so it's right, but I had to think about that

File: atlasdb-api/build.gradle [hunk 3]
```diff
@@ -4,6 +4,8 @@ apply plugin: 'com.palantir.revapi'
 dependencies {
   api project(":lock-api")
   api project(":timestamp-api")
+  api 'com.palantir.conjure.java.api:service-config'
```
Comment (jeremyk-91) [inline atlasdb-api/build.gradle]: Might be missing something: I'm guessing this is required because of dialogue clients?

**Review by jeremyk-91 — DISMISSED** [review]
👍  just one question about dependencies

File: atlasdb-api/build.gradle (hunk 3 above)
Comment (jkozlowski) [reply atlasdb-api/build.gradle]: ah, sorry, I'll remove. I initially also plumbed UserAgents, until I realized that's redundant

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java (hunk 1 above)
Comment (jkozlowski) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:416]: +1 fwiw this is exactly my reasoning, it felt like one less PR.

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java (hunk 2 above)
Comment (jkozlowski) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:628]: Yeah, I suspect this is remoting -> conjure.

File: atlasdb-api/build.gradle (hunk 3 above)
Comment (jkozlowski) [reply atlasdb-api/build.gradle]: Dne

**Review by jeremyk-91 — APPROVED** [review]
👍
